scripts/common.py

```diff
@@ -1023,3 +1032,286 @@ def managed_cloud_datastore_emulator(clear_datastore=False):
             'DATASTORE_USE_PROJECT_ID_AS_APP_ID', 'true'))
 
         yield proc
+
+
+@contextlib.contextmanager
+def managed_redis_server():
+    """Run the redis server within a context manager that ends it gracefully."""
+    if is_windows_os():
+        raise Exception(
+            'The redis command line interface is not installed because your '
+            'machine is on the Windows operating system. The redis server '
+            'cannot start.')
+
+    # Check if a redis dump file currently exists. This file contains residual
+    # data from a previous run of the redis server. If it exists, removes the
+    # dump file so that the redis server starts with a clean slate.
+    if os.path.exists(REDIS_DUMP_PATH):
+        os.remove(REDIS_DUMP_PATH)
+
+    # Start the redis local development server. Redis doesn't run on
+    # Windows machines.
+    proc_context = managed_process(
+        [REDIS_SERVER_PATH, REDIS_CONF_PATH], title='Redis Server', shell=True)
+    with proc_context as proc:
+        wait_for_port_to_be_in_use(feconf.REDISPORT)
+        yield proc
+
+
+def create_managed_web_browser(port):
+    """Returns a context manager for a web browser targeting the given port on
+    localhost. If a web browser cannot be opened on the current system by Oppia,
+    then returns None instead.
+
+    Args:
+        port: int. The port number to open in the web browser.
+
+    Returns:
+        context manager|None. The context manager to a web browser window, or
+        None if the current operating system does not support web browsers.
+    """
+    url = 'http://localhost:%s/' % port
+    title = 'Web Browser'
+    if is_linux_os():
+        if any(re.match('.*VBOX.*', d) for d in os.listdir('/dev/disk/by-id/')):
+            return None
+        else:
+            return managed_process(['xdg-open', url], title=title)
+    elif is_mac_os():
+        return managed_process(['open', url], title=title)
+    else:
+        return None
+
+
+@contextlib.contextmanager
+def managed_webpack_compiler(
+        config_path=None, use_prod_env=False, use_source_maps=False,
+        watch_mode=False, max_old_space_size=None):
+    """Returns context manager to start/stop the webpack compiler gracefully.
+
+    Args:
+        config_path: str|None. Path to an explicit webpack config, or None to
```

Comment:
This branching seems not so nice, can we just do one or the other?

Reply:
The other option would be a similar branch (if use_terser_config, config_path = terser_config_path, else if prod_env ...), so I don't think there's much to gain.